core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1139,6 +1160,7 @@ class ReplicaManager(val config: KafkaConfig,
       } else {
         val deletedPartitions = metadataCache.updateMetadata(correlationId, updateMetadataRequest)
         controllerEpoch = updateMetadataRequest.controllerEpoch
+        followerReassignmentLagStats.removeStats(deletedPartitions)
```

Comment:
I am not sure it makes sense to do this in response to metadata updates. Seems like `stopReplica` would be better option. Note that is where we call `maybeRemoveTopicMetrics`. Probably it makes sense to remove the reassignment metrics as part of `maybeRemoveTopicMetrics`?